### PR TITLE
<2.7> Convert test_wl_upgrade to go

### PR DESCRIPTION
**Pull request by lscalabrini01**
Implemented TestWorkloadUpgrade test

### CODE REVIEW

**Review by floatingman — APPROVED** [review]
LGTM

**Review by floatingman — APPROVED** [review]
LGTM

File: tests/v2/actions/workloads/pods/pods.go [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
 const (
-	timeFormat = "2006/01/02 15:04:05"
-	imageName  = "nginx"
+	timeFormat   = "2006/01/02 15:04:05"
+	imageName    = "nginx"
+	podSteveType = "pod"
```
Comment (slickwarren) [inline tests/v2/actions/workloads/pods/pods.go:28]: missing from other PRs: 
```
historyHeaderLength = 2
	revisionNumberIndex = 0
	historyHeader       = "REVISION  CHANGE-CAUSE"
	revisionsIndex      = 1
	revisionAnnotation  = "deployment.kubernetes.io/revision"
```

File: tests/v2/actions/workloads/pods/pods.go [hunk 2]
```diff
@@ -66,3 +83,127 @@ func NewPodTemplateWithConfig(secretName, configMapName string, useEnvVars, useV
 	containers := []corev1.Container{container}
 	return workloads.NewPodTemplate(containers, volumes, nil, nil, nil)
 }
+
+// CheckPodLogsForErrors is a helper to check pod logs for errors
+func CheckPodLogsForErrors(client *rancher.Client, clusterID string, podName string, namespace string, errorPattern string, startTime time.Time) error {
```
Comment (slickwarren) [inline tests/v2/actions/workloads/pods/pods.go:88]: net new in 2.7 vs. 2.8/2.9/main. Can you confirm this isn't an extension on 2.7 for some reason?

**Review by slickwarren — COMMENTED** [review]
some questions about the line count diff compared to 2.8

File: tests/v2/actions/workloads/pods/pods.go (hunk 1 above)
Comment (lscalabrini01) [reply tests/v2/actions/workloads/pods/pods.go:28]: It is not missing, it is in [deployment.go file](https://github.com/rancher/rancher/pull/46875/files#diff-b6843bc14cef727d6cded1a32365e9be967b17b84a7add1b3313d3d48efb4e45R26)

File: tests/v2/actions/workloads/pods/pods.go (hunk 2 above)
Comment (lscalabrini01) [reply tests/v2/actions/workloads/pods/pods.go:88]: Hi, this function is on 2.8/2.9/main, but it is just used on 2.9/main, I don't know why it is not an extension on 2.7, I'll remove this function

**Review by slickwarren — APPROVED** [review]

**Review by floatingman — APPROVED** [review]
LGTM

**Pull request merged**